### PR TITLE
Improve post locking modal

### PR DESCRIPTION
**Pull request by dimadin**
There are few issues with post locking modal introduced in #4217. This PR tries to fix them and make it look more like post locking modal in classic editor screen (see function [`_admin_notice_post_locked()`](https://developer.wordpress.org/reference/functions/_admin_notice_post_locked/)).

### Current post type

There are two changes here: "All Posts" and "View all posts" texts were changed to use current post type label, and URL of those links now links to the screen of current post type. Both of these changes already exist in post locking modal in classic editor screen.

### Translators comments

Explanation that "post" is generic and may be of any type is removed. There are two reasons for this: there are other strings using "post" in this modal that don't have such comment and they shouldn't have it because those strings can be used elsewhere (they would need context instead of comment), and this is how it's done in post locking modal in classic editor screen.

### Styling

Link to "All Posts" screen was changed to button for consistency with both other case and usage in post locking modal in classic editor screen.

### i18n

Added translator comments for strings with placeholder, added dot at the end of two sentences that didn't have it, and used apostrophe instead of single-quote character as per [guidelines](https://github.com/WordPress/gutenberg/blob/master/docs/reference/coding-guidelines.md#strings).

### CODE REVIEW

File: packages/editor/src/components/post-locked-modal/index.js [hunk 1]
```diff
@@ -155,38 +159,37 @@ class PostLockedModal extends Component {
 						<div>
 							{ userDisplayName ?
 								sprintf(
-									/* translators: 'post' is generic and may be of any type (post, page, etc.). */
-									__( '%s now has editing control of this post. Don\'t worry, your changes up to this moment have been saved' ),
+									/* translators: %s: user's display name */
+									__( '%s now has editing control of this post. Don’t worry, your changes up to this moment have been saved.' ),
```
Comment (adamsilverstein) [inline packages/editor/src/components/post-locked-modal/index.js]: I think we want to keep the slashed single quote, the there a reason that was changed here?
```suggestion
									__( '%s now has editing control of this post. Don\'t worry, your changes up to this moment have been saved.' ),
```

File: packages/editor/src/components/post-locked-modal/index.js [hunk 2]
```diff
@@ -155,38 +159,37 @@ class PostLockedModal extends Component {
...
 									userDisplayName
 								) :
-								/* translators: 'post' is generic and may be of any type (post, page, etc.). */
-								__( 'Another user now has editing control of this post. Don\'t worry, your changes up to this moment have been saved' )
+								__( 'Another user now has editing control of this post. Don’t worry, your changes up to this moment have been saved.' )
```
Comment (adamsilverstein) [inline packages/editor/src/components/post-locked-modal/index.js]:
```suggestion
								__( 'Another user now has editing control of this post. Don\'t worry, your changes up to this moment have been saved.' )
```

**Comment by adamsilverstein** [comment]
@dimadin thank you for the contribution here. overall this looks good, I am not certain about the translator comments - your explanation makes sense. 

I left a comment about using escaped single quotes before seeing your link to the guidelines - thanks for that - which are new to me and don't quite match these strings in core and what we have for core standards (https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#strings, https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#single-and-double-quotes) - my only (possibly unfounded) concern would be that these characters would somehow not render correctly in certain fonts or languages.

**Comment by adamsilverstein** [comment]
@dimadin thanks for clarify, looks good to me!

**Comment by adamsilverstein** [comment]
Tested this out locally and looks good. One potential issue using the full post type label for the button is that for longer post type names, the current fixed width design may lead to some button overflow that doesn't look great. 

Somewhat of an edge case, this is probably something we can fix by adjusting the CSS in a subsequent PR.

**Review by adamsilverstein — APPROVED** [review]

**Comment by gziolo** [comment]
> Somewhat of an edge case, this is probably something we can fix by adjusting the CSS in a subsequent PR.

Let's make sure that it's tracked before we merge this PR.

File: packages/editor/src/components/post-locked-modal/index.js (hunk 1 above)
Comment (tofumatt) [reply packages/editor/src/components/post-locked-modal/index.js]: An apostrophe is the correct character to use in English here, not a single quote. It's certainly typographically better; that said, we should just be consistent and use one or the other everywhere. See: #7555.

**Comment by ocean90** [comment]
> Unfortunately, GlotPress search doesn't work with `’`

@dimadin It works if you do a [case sensitive search](https://translate.wordpress.org/projects/wp-plugins/gutenberg/stable/sr/default?filters%5Bterm%5D=%E2%80%99&filters%5Buser_login%5D=&filters%5Bstatus%5D=current_or_waiting_or_fuzzy_or_untranslated&filters%5Bcase_sensitive%5D=yes&filter=Filter&sort%5Bby%5D=priority&sort%5Bhow%5D=desc) as it performs binary comparisons.

**Comment by mtias** [comment]
Thanks for the improvements!

**Pull request merged**